packages/compass-explain-plan/src/modules/explain.js

```diff
@@ -320,16 +348,25 @@ export const fetchExplainPlan = (query) => {
       return;
     }
 
+    const abortController = new AbortController();
```

Comment:
Before starting a new fetch, you probably want to check whether another one is already running. Otherwise you can put this tab in a weird state by clicking "Explain" button multiple times

Reply:
Yea you're right. I just noticed that we don't have a disabled state for button for when the operation is running. In that case, I guess we can assume that user would like the result for his last click and hence we just cancel the previous explain.

Reply:
changes implemented